src/databricks/labs/lsql/backends.py

```diff
@@ -177,23 +156,45 @@ def _row_to_sql(cls, row: DataclassInstance, fields: tuple[dataclasses.Field[Any
         data = []
         for f in fields:
             value = getattr(row, f.name)
-            field_type = cls._field_type(f)
-            if isinstance(field_type, UnionType):
-                field_type = field_type.__args__[0]
-            if value is None:
-                data.append("NULL")
-            elif field_type is bool:
-                data.append("TRUE" if value else "FALSE")
-            elif field_type is str:
-                value = str(value).replace("'", "''")
-                data.append(f"'{value}'")
-            elif field_type is int:
-                data.append(f"{value}")
-            else:
-                msg = f"unknown type: {field_type}"
-                raise ValueError(msg)
+            data.append(cls._value_to_sql(value))
         return ", ".join(data)
 
+    @classmethod
+    def _value_to_sql(cls, value: Any) -> str:
+        if value is None:
+            return "NULL"
+        if isinstance(value, int):
+            return f"{value}"
+        if isinstance(value, float):
+            return f"{value:0.2f}"
+        if isinstance(value, str):
+            value = str(value).replace("'", "''")
+            return f"'{value}'"
+        if isinstance(value, bool):
+            return "TRUE" if value else "FALSE"
+        if isinstance(value, datetime.date):
+            return f"'{value.year}-{value.month}-{value.day}'"
+        if isinstance(value, datetime.datetime):
+            return f"'{value.strftime('%Y-%m-%d %H:%M:%S')}'"
+        if isinstance(value, list):
+            values = ", ".join(cls._value_to_sql(v) for v in value)
+            return f"ARRAY({values})"
+        if isinstance(value, dict):
```

Comment:
As with `list` should we allow non-dict mappings here? (Named tuples and mapping-proxy spring to mind.)

Reply:
no, because we can't infer types from those just yet